src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -3190,5 +3060,102 @@ subroutine convert_cvpnx_to_nx(qnx_arr,cvpnr,cvpnr_pvalue,cloud_nt_updt,q_arr,qr
     qnx_arr=tmparr_qnr
 
 end subroutine convert_cvpnx_to_nx
+   subroutine gsi_copy_bundle(bundi,bundo) 
+   use gsi_bundlemod, only:gsi_bundleinquire, gsi_bundlegetpointer,gsi_bundleputvar
+!ct   use gsi_bundlemod, only : gsi_bundle
+   implicit none  
+    
+!  copy the variables in the gsi_metguess_bundle_inout to gsi_bundle_inout or
+!  vice versa, according to icopy_flag  
+! !INPUT PARAMETERS:
+
+   type(gsi_bundle), intent(in   ) :: bundi
+
+! !INPUT/OUTPUT PARAMETERS:
+
+   type(gsi_bundle), intent(inout) :: bundo
+   character(len=max_varname_length),dimension(:),allocatable:: src_name_vars2d
+   character(len=max_varname_length),dimension(:),allocatable:: src_name_vars3d
+   character(len=max_varname_length),dimension(:),allocatable:: target_name_vars2d
+   character(len=max_varname_length),dimension(:),allocatable:: target_name_vars3d
+   character(len=max_varname_length) ::varname 
+   real(r_kind),dimension(:,:,:),pointer:: pvar3d
+   real(r_kind),dimension(:,:,:),pointer:: pvar2d
+   integer(i_kind):: src_nc3d,src_nc2d,target_nc3d,target_nc2d
+   integer(i_kind):: ivar,jvar,istatus
+   src_nc3d=bundi%n3d
+   src_nc2d=bundi%n2d
+   target_nc3d=bundo%n3d
+   target_nc2d=bundo%n2d
+   allocate(src_name_vars3d(src_nc3d),src_name_vars2d(src_nc2d))
+   allocate(target_name_vars3d(target_nc3d),target_name_vars2d(target_nc2d))
+   call gsi_bundleinquire(bundi,'shortnames::3d',src_name_vars3d,istatus)
+   call gsi_bundleinquire(bundi,'shortnames::2d',src_name_vars2d,istatus)
+   call gsi_bundleinquire(bundo,'shortnames::3d',target_name_vars3d,istatus)
+   call gsi_bundleinquire(bundo,'shortnames::2d',target_name_vars2d,istatus)
+   do ivar=1,src_nc3d
+      varname=trim(src_name_vars3d(ivar))
+      do jvar=1,target_nc3d
+        if(index(target_name_vars3d(jvar),varname).gt.0)  then
+        call GSI_BundleGetPointer (bundi,varname,pvar3d,istatus)
+        call gsi_bundleputvar (bundo,varname,pvar3d,istatus)
+!         write(6,*)varname ,' is copied from bundi to bundo in  gsi_copy_bundle'
+        exit
+        endif
```

Comment:
Please replace .gt. with >, if line 3102 won't be used in the future, please remove it, and please apply proper indentation to lines 3099 through 3104.